packages/compass-aggregations/src/components/stage-toolbar/index.tsx

```diff
@@ -158,8 +162,15 @@ type StageToolbarOwnProps = Pick<
 
 export default connect(
   (state: RootState, ownProps: StageToolbarOwnProps) => {
-    const stage = state.pipelineBuilder.stageEditor.stages[ownProps.index];
+    const {
+      pipelineBuilder: {
+        stageEditor: { stages },
+      },
+    } = state;
+    const stage = stages[ownProps.index];
+    assertReduxStage(stage);
     return {
+      idxInPipeline: storeIndexToPipelineIndex(stages, ownProps.index),
```

Comment:
Wouldn't this always be same as `ownProps.index`? We are rendering all the stage components (preview, editor and toolbar) inside a stage which is rendered only for ReduxStage.

Reply:
It won't remain the same as soon as we have Wizards in the state. With wizards the index will shift but because we want to render the stage titles with their actual position, we calculate this.

I did a recent change, where we don't calculate but simply takes it from stage.